blazor/datagrid/filtering.md

```diff
@@ -7,108 +7,422 @@ control: DataGrid
+    {
+        GridData = OrderData.GetAllRecords();
+        ApplyInitialFilter();
+    }
+ 
+    private void ApplyInitialFilter()
+    {
+        var Col1Pre = new WhereFilter();
+        var predicate = new List<WhereFilter>();
+        predicate.Add(new WhereFilter() {Field= "CustomerID",
+                Operator = "startswith",
+                value = "VINET", });
+        predicate.Add(new WhereFilter()
+            {
+                Field = "CustomerID",
+                Operator = "startswith",
+                value = "HANAR",
+            });
+        Col1Pre = WhereFilter.Or(predicate);
+        Qry = new Query().Where(Col1Pre);
+    }
+}
+{% endhighlight %}
+{% highlight c# tabtitle="OrderData.cs" %}
+ public class OrderData
     {
+        public static List<OrderData> Orders = new List<OrderData>();
+ 
+        public OrderData()
+        {
+ 
+        }
+        public OrderData(int? OrderID,string CustomerID,string ShipCity, string ShipName)
+        {
+            this.OrderID = OrderID;    
+            this.CustomerID = CustomerID;
+            this.ShipCity = ShipCity;
+            this.ShipName = ShipName;           
+        }
+ 
+        public static List<OrderData> GetAllRecords()
+        {
+            if (Orders.Count() == 0)
+            {
+                int OrderID = 10248;
+ 
+                for (int i = 1; i < 3; i++)
+                {
+                    Orders.Add(new OrderData(OrderID+1, "VINET", "Reims", "Vins et alcools Chevali"));
+                    Orders.Add(new OrderData(OrderID+2, "TOMSP", "Münster", "Toms Spezialitäten"));
+                    Orders.Add(new OrderData(OrderID+3, "HANAR", "Rio de Janeiro", "Hanari Carnes"));
+                    Orders.Add(new OrderData(OrderID+4, "VICTE", "Lyon", "Victuailles en stock"));
+                    Orders.Add(new OrderData(OrderID+5, "SUPRD", "Charleroi", "Suprêmes délices"));
+                    Orders.Add(new OrderData(OrderID+6, "HANAR", "Lyon", "Hanari Carnes"));
+                    Orders.Add(new OrderData(OrderID+7, "CHOPS", "Rio de Janeiro", "Chop-suey Chinese"));
+                    Orders.Add(new OrderData(OrderID+8, "RICSU", "Münster", "Richter Supermarkt"));
+                    Orders.Add(new OrderData(OrderID+9, "WELLI", "Reims", "Wellington Import"));
+                    OrderID += 9;
+                }
+            }
+            return Orders;
+        }
+ 
         public int? OrderID { get; set; }
         public string CustomerID { get; set; }
-        public DateTime? OrderDate { get; set; }
-        public double? Freight { get; set; }
+        public string ShipCity { get; set; }
+        public string ShipName { get; set; }
+   }
+{% endhighlight %}
+{% endtabs %}
+
+{% previewsample "https://blazorplayground.syncfusion.com/embed/rXhTjMilotUqJJMF?appbar=false&editor=false&result=true&errorlist=false&theme=bootstrap5" %}
+
+### Initial filter with multiple values for different columns using Query property
```

Comment:
Try this topic with this solution. 
https://blazor.syncfusion.com/documentation/datagrid/filter-menu#filtering-using-daterangepicker - filtersettings columns. remove query property and change content if this solution works.